clang/lib/CIR/Dialect/Transforms/HoistAllocas.cpp

```diff
@@ -0,0 +1,66 @@
+//===- HoistAllocas.cpp - performs CIR simplification ----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetail.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "clang/CIR/Dialect/IR/CIRDialect.h"
+#include "clang/CIR/Dialect/Passes.h"
+
+using namespace mlir;
+using namespace mlir::cir;
+
+namespace {
+
+struct HoistAllocasPass : public HoistAllocasBase<HoistAllocasPass> {
+  HoistAllocasPass() = default;
+  void runOnOperation() override;
+};
+
+static void process(mlir::cir::FuncOp func) {
+  if (func.empty())
+    return;
+
+  std::vector<mlir::cir::AllocaOp> allocas;
+
+  mlir::Block &entry = *func.begin();
+
+  func.getBody().walk([&](mlir::Operation *op) {
+    auto alloca = dyn_cast<mlir::cir::AllocaOp>(op);
+    if (!alloca)
+      return;
```

Comment:
I believe this could be simplified to just:

```suggestion
  func.getBody().walk([&](mlir::cir::AllocaOp op) {
```

Reply:
Done